src/sources/soundsourceffmpeg.cpp

```diff
@@ -304,672 +416,842 @@ SoundSourceFFmpeg::~SoundSourceFFmpeg() {
+                const auto clearSampleCount =
+                        frames2samples(missingFrameCount);
+                const SampleBuffer::WritableSlice writableSlice(
+                        m_sampleBuffer.growForWriting(clearSampleCount));
+                DEBUG_ASSERT(writableSlice.length() == clearSampleCount);
+                SampleUtil::clear(
+                        writableSlice.data(),
+                        clearSampleCount);
+                readFrameIndex += missingFrameCount;
+            }
+            if (!decodedFrameRange.empty()) {
+                const auto copySampleCount =
+                        frames2samples(decodedFrameRange.length());
+                const SampleBuffer::WritableSlice writableSlice(
+                        m_sampleBuffer.growForWriting(copySampleCount));
+                DEBUG_ASSERT(writableSlice.length() == copySampleCount);
+                SampleUtil::copy(
+                        writableSlice.data(),
+                        pDecodedSampleData,
+                        copySampleCount);
+                readFrameIndex += decodedFrameRange.length();
+            }
 
-        m_bIsSeeked = m_currentMixxxFrameIndex != frameIndexMin();
+            // Housekeeping before next decoding iteration
+            av_frame_unref(m_pavDecodedFrame);
+            av_frame_unref(m_pavResampledFrame);
+        } while ((avcodec_receive_frame_result == 0) && (m_curFrameIndex != kFrameIndexInvalid));
     }
-    DEBUG_ASSERT(m_currentMixxxFrameIndex == seekFrameIndex);
-
-    const SINT numberOfFrames = writableSampleFrames.frameLength();
-
-    DEBUG_ASSERT(m_currentMixxxFrameIndex == firstFrameIndex);
-    DEBUG_ASSERT(m_SCache.size() > 0);
-    getBytesFromCache(
-            writableSampleFrames.writableData(),
-            m_currentMixxxFrameIndex,
-            numberOfFrames);
-    m_currentMixxxFrameIndex += numberOfFrames;
-    m_bIsSeeked = false;
+    DEBUG_ASSERT(!pavNextPacket);
 
+    const auto readableRange =
+            IndexRange::between(
+                    writableSampleFrames.frameIndexRange().start(),
+                    writableRange.start());
     return ReadableSampleFrames(
-            IndexRange::forward(firstFrameIndex, numberOfFrames),
+            readableRange,
             SampleBuffer::ReadableSlice(
                     writableSampleFrames.writableData(),
-                    std::min(writableSampleFrames.writableLength(), frames2samples(numberOfFrames))));
+                    frames2samples(readableRange.length())));
+}
+
+QString SoundSourceProviderFFmpeg::getName() const {
+    return "FFmpeg";
+}
+
+SoundSourceProviderPriority SoundSourceProviderFFmpeg::getPriorityHint(
+        const QString& /*supportedFileExtension*/) const {
+    // TODO: Increase priority to HIGHER if FFmpeg should be used as the
+    // default decoder instead of other SoundSources. Currently it is
+    // only used as a fallback.
+    return SoundSourceProviderPriority::LOWER;
```

Comment:
I think this should be LOWEST before merge, right?

Reply:
Why? All tests pass.

Reply:
Ok, let's use LOWEST, because we use LOWER for libsndfile.